std/format.d

```diff
@@ -8403,3 +8449,696 @@ printFloat_done:
     f.precision = 1;
     assert(printFloat(-245.666f, f) == "-2.5E+02");
 }
+
+private auto printFloatF(T, Char)(T val, FormatSpec!Char f, RoundingMode rm,
+                                  string sgn, int exp, ulong mnt, bool is_upper)
+if (is(T == float) || is(T == double) || (is(T == real) && T.mant_dig == double.mant_dig))
+{
+    import std.conv : to;
+    import std.algorithm.comparison : max;
+
+    enum int bias = T.max_exp - 1;
+
+    if (f.precision == f.UNSPECIFIED)
+        f.precision = 6;
+
+    // special treatment for 0.0
+    if (exp == 0 && mnt == 0)
```

Comment:
Can you factor this branch out into a separate function and merge the logic with the `0.0` case for `printFloatE`. They look almost identical, modulo some minor handling for the difference in the format specifiers.

Reply:
done.

Reply:
Thanks!